lambdas/src/compile.rs

```diff
@@ -0,0 +1,177 @@
+use aws_config::BehaviorVersion;
+use aws_sdk_dynamodb::{error::SdkError, operation::put_item::PutItemError};
+use lambda_http::{
+    run, service_fn, Error as LambdaError, Request as LambdaRequest, Response as LambdaResponse,
+};
+use serde::Deserialize;
+use std::ops::Add;
+use tracing::{error, info};
+
+mod common;
+use crate::common::{errors::Error, utils::extract_request, Item, Status, BUCKET_NAME_DEFAULT};
+
+const QUEUE_URL_DEFAULT: &str = "https://sqs.ap-southeast-2.amazonaws.com/266735844848/zksync-sqs";
+const TABLE_NAME_DEFAULT: &str = "zksync-table";
+
+const NO_OBJECTS_TO_COMPILE_ERROR: &str = "There are no objects to compile";
+
+#[derive(Debug, Deserialize)]
+struct Request {
+    pub id: String,
+}
+
+// impl Deserialize for Response {
+//     fn deserialize<'de, D>(deserializer: D) -> Result<Self, D::Error> where D: Deserializer<'de> {
+//         todo!()
+//     }
+// }
+// TODO:
+// struct SqsClient {
+//     pub client: aws_sdk_sqs::Client,
+//     pub queue_url: String,
+//    // pub other_data: String
+// }
+
+async fn compile(
+    id: String,
+    dynamo_client: &aws_sdk_dynamodb::Client,
+    table_name: &str,
+    sqs_client: &aws_sdk_sqs::Client,
+    queue_url: &str,
+) -> Result<(), Error> {
+    let item = Item {
+        id: id.clone(),
+        status: Status::Pending,
+    };
+
+    let result = dynamo_client
+        .put_item()
+        .table_name(table_name)
+        .set_item(Some(item.into()))
+        .condition_expression("attribute_not_exists(ID)")
+        .send()
+        .await;
+
+    match result {
+        Ok(val) => val,
+        Err(SdkError::ServiceError(val)) => match val.err() {
+            PutItemError::ConditionalCheckFailedException(_) => {
+                error!("Recompilation attempt, id: {}", id);
+                let response = lambda_http::Response::builder()
+                    .status(400)
+                    .header("content-type", "text/html")
+                    .body("Recompilation attempt".into())
```

Comment:
Let's make "recompilation attemp" const too, as done for `QUEUE_URL_DEFAULT`?

Reply:
done